packages/graph/hash_graph/lib/graph/src/api/web/error.rs

```diff
@@ -0,0 +1,37 @@
+#![allow(
+    clippy::needless_pass_by_value,
+    reason = "This is considered the last-stop for errors from the perspective of the web-api and \
+              taking Owned reports allows us to simply `map_err` and return the appropriate error \
+              response from request handlers"
+)]
+
+use axum::http::StatusCode;
+use error_stack::Report;
+
+use crate::datastore::{BaseIdAlreadyExists, BaseIdDoesNotExist, QueryError};
+
+/// Convert an error report to an HTTP status code depending on values provided.
+/// Assumes the report is created by a _write_ operation in the datastore.
+pub fn modify_report_to_status_code<T>(report: Report<T>) -> StatusCode {
```

Comment:
These errors do make a lot of assumptions about the caller, and the errors lack any information for the user. We can consider this the first version, and add user-facing errors when we figure out how we want to specify them